screens/ReservationUpcomingForm.js

```diff
@@ -104,29 +129,237 @@ class ReservationUpcomingForm extends React.Component {
+      this.handleCheckAvailableTables(event.id, event.reservationStartDate)
+    }
+  }
+  handleCheckAvailableTables = (id, time) => {
+
+    let checkDate = moment(time).format('YYYY-MM-DDTHH:mm:ss')
+
+    console.log("check date", checkDate)
+    dispatchFetchRequestWithOption(api.reservation.getAvailableTables(checkDate, id), {
+      method: 'GET',
+      withCredentials: true,
+      credentials: 'include',
+      headers: {},
+    }, {
+      defaultMessage: false
+    },
+      response => {
+        response.json().then(data => {
+          this.setState({availableTables: data.results, isGetTables: true})
+
+        })
+      }).then()
+
+  }
+  handleUpdateReservation = (values) => {
+    let request = {
+      reservationDate: moment(values.reservationStartDate).format('YYYY-MM-DDTHH:mm:ss'),
+      name: values.name,
+      phoneNumber: values.phoneNumber,
+      tableIds: this.state.selectedTableIds,
+      people: values?.people,
+      kid: values?.kid,
+      note: values?.note
+    }
+    console.log("check update=", request)
+
+    if (this.state.selectedTableIds && this.state.selectedTableIds.length > 0) {
+      dispatchFetchRequestWithOption(
+        api.reservation.update(values.id),
+        {
+          method: 'POST',
+          withCredentials: true,
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(request)
+        }, {defaultMessage: true},
+        response => {
+          this.handleToggleTableList(false, null)
+          this.setState({selectedTableIds: []})
+        }
+      ).then(
+        setTimeout(() => {
+          this.refreshScreen()
+        }, 300)
+      )
+    } else {
+      Alert.alert(
+        ``,
+        `${this.context.t('reservation.selectTablesContext')}`,
+        [
+          {
+            text: `${this.context.t('action.confirm')}`,
+            onPress: () => {}
+          }
+        ]
+      )
+    }
+  }
+
+  handleChooseTable = (id, name) => {
+    let tableIds = this.state.selectedTableIds
+    let cancelTableIdx = tableIds.indexOf(id)
+
+    if (!!tableIds && cancelTableIdx > -1) {
+      tableIds.splice(cancelTableIdx, 1)
+      this.setState({selectedTableIds: tableIds})
+    } else {
+      this.setState({selectedTableIds: [...tableIds, id]})
+    }
+  }
+
+
+  handleToggleMobileForm = (flag) => {
+    this.setState({mobileFormVisible: flag})
+  }
+
+
 
   render() {
     const {
       navigation,
       route,
+      haveData,
+      isLoading,
+      tablelayouts,
+      ordersInflight,
+      availableTables,
       handleSubmit,
     } = this.props
 
     const {t, customMainThemeColor, customBackgroundColor} = this.context
     const timeBlocks = Object.keys(this.state.timeBlocks).map(key => this.state.timeBlocks[key].label)
     const timezone = TimeZoneService.getTimeZone()
 
-    const testView = true
+    const testView = false
 
 
     if (testView) {
```

Comment:
What is testView for?

Reply:
It's for "Coming soon" display. Now I think can remove it.